crates/bevy_ecs/src/component.rs

```diff
@@ -194,10 +194,15 @@ impl ComponentDescriptor {
         }
     }
 
-    pub fn new_resource<T: Resource>(storage_type: StorageType) -> Self {
+    /// Create a new `ComponentDescriptor` for a resource.
+    ///
+    /// The [`StorageType`] for resources is always [`TableStorage`].
+    pub fn new_resource<T: Resource>() -> Self {
         Self {
             name: std::any::type_name::<T>().to_string(),
-            storage_type,
+            // NOTE: `SparseStorage` may actually be a more
```

Comment:
```suggestion
            // PERF: `SparseStorage` may actually be a more
```

This will be a bit more discoverable, and this is fundamentally a change that would need to be benchmarked.